packages/zk-eddsa-event-ticket-pcd/src/ZKEdDSAEventTicketPCD.ts

```diff
@@ -53,7 +54,7 @@ export const STATIC_TICKET_PCD_NULLIFIER = generateSnarkMessageHash(
 export const ZKEdDSAEventTicketPCDTypeName = "zk-eddsa-event-ticket-pcd";
 
 let depsInitializedPromise: Promise<void> | undefined;
-let babyJub: BabyJub;
+let bn128: any; // This curve type comes from ffjavascript, which doesn't provide types.
```

Comment:
I honestly don't know what Montgomery conversion is.  I think we stopped using that form, but honestly the various format conversions here are black magic to me, and I don't know if they can be replaced.

Testing indicates I can indeed use `eddsa.F` instead of `bn128.Fr`.  With that change, there are only 3 lines in `snarkInputForProof` where the `eddsa` object is used, and they're related to each other:

    const rawSig = eddsa.unpackSignature(
      fromHexString(ticketPCD.proof.eddsaPCD.proof.signature)
    );
    // ...
    ticketSignatureR8x: eddsa.F.toObject(rawSig.R8[0]).toString(),
    ticketSignatureR8y: eddsa.F.toObject(rawSig.R8[1]).toString(),

Can you see a way these lines can be simplified or performed without the need for wasm curves?

Reply:
`F.toObject` should convert a value from Montgomery format to a standard one. I don't know how it works specifically either but I think that's not important here.

I think there's no way to further simplify/perform it :/ We need to wait for https://github.com/iden3/ffjavascript/pull/130.

Reply:
I thought there was a PR about eliminating Montgomery form, which resulted in these deps being removed from verify, and now being used only in prove.  @gubsheep do you have any context on whether there's more work to be done there?  Should we be changing the format of the signature in the ticket PCD?

Reply:
> I thought there was a PR about eliminating Montgomery form, which resulted in these deps being removed from verify, and now being used only in prove. @gubsheep do you have any context on whether there's more work to be done there? Should we be changing the format of the signature in the ticket PCD?

Oh ye I've worked on that -> https://github.com/proofcarryingdata/zupass/pull/663. We shouldn't need to change anything here as `F` is still the same object and it converts those values as before.

Reply:
If I'm understanding correctly, #663 changed the storage of the EdDSA pub key from Mongomery form to standard.  It didn't change the storage of the EdDSA signature, which is what still needs conversion here.  I think my question is, would it be possible or desirable to change the storage format of the signature too, so that we don't need these kinds of conversions to make use of it in SNARKs.

Reply:
> If I'm understanding correctly, #663 changed the storage of the EdDSA pub key from Mongomery form to standard. It didn't change the storage of the EdDSA signature, which is what still needs conversion here. I think my question is, would it be possible or desirable to change the storage format of the signature too, so that we don't need these kinds of conversions to make use of it in SNARKs.

Oh got it! Let's wait for @gubsheep for confirmation, but I don't think we can change the format when it's packed. If we can we should also consider packing the pubkey with https://github.com/iden3/circomlibjs/blob/4f094c5be05c1f0210924a3ab204d8fd8da69f49/src/babyjub.js#L97 imo.

Reply:
Poked @gubsheep on Discord to see if he can take a look, but I think the result of that conversation can/should be a separate follow-up.  This PR in its current form simply reduces dependencies, so I think it should merge as-is.

Reply:
> This PR in its current form simply reduces dependencies, so I think it should merge as-is.

agree, this PR in current form looks fine.

> If I'm understanding correctly, https://github.com/proofcarryingdata/zupass/pull/663 changed the storage of the EdDSA pub key from Mongomery form to standard. It didn't change the storage of the EdDSA signature, which is what still needs conversion here. I think my question is, would it be possible or desirable to change the storage format of the signature too, so that we don't need these kinds of conversions to make use of it in SNARKs.

Yes, it would be possible + desirable. Basically we would want to move the logic that is currently performed in ZKEdDSAEventTicketPCD [here](https://github.com/proofcarryingdata/zupass/blob/488bfda1360c6094df1096d6da598da47db1379a/packages/zk-eddsa-event-ticket-pcd/src/ZKEdDSAEventTicketPCD.ts#L278) and [here](https://github.com/proofcarryingdata/zupass/blob/488bfda1360c6094df1096d6da598da47db1379a/packages/zk-eddsa-event-ticket-pcd/src/ZKEdDSAEventTicketPCD.ts#L325), into the [prove function](https://github.com/proofcarryingdata/zupass/blob/488bfda1360c6094df1096d6da598da47db1379a/packages/eddsa-pcd/src/EdDSAPCD.ts#L151) of EdDSAPCD

Reply:
Is an unpacked signature the best format for use cases which don't involve generating a ZK proof?  It seems like verifying the signature does expect Montgomery form.  I'm wondering if by switching we're making our PCD too specific to our use case.

The next question is, can we make such a change fast enough, and safely enough.  There are users with signed tickets already, or there soon will be.  If we change the ticket format, we need to make sure their tickets get re-signed (does the feed do that by default?) and/or add a version to the PCD type.

Reply:
re: first paragraph - good point.

so it looks like, at the very least, `eddsa.unpackSignature(fromHexString(signature))` is something that both EdDSAPCD [verify](https://github.com/proofcarryingdata/zupass/blob/488bfda1360c6094df1096d6da598da47db1379a/packages/eddsa-pcd/src/EdDSAPCD.ts#L167) and ZKEdDSA [snark args generation](https://github.com/proofcarryingdata/zupass/blob/488bfda1360c6094df1096d6da598da47db1379a/packages/zk-eddsa-event-ticket-pcd/src/ZKEdDSAEventTicketPCD.ts#L278) need us to do. And as best as I can tell, [`unpackSignature`](https://github.com/iden3/circomlibjs/blob/main/src/eddsa.js#L277) gives us an object containing a point with BigInt coordinates in Montgomery form, and another BigInt.

however, given that eddsa verify does expect the signature in montgomery form, i'm no longer sure if representing the signature in edwards form as the default representation in the PCD is in fact desirable. 

========

regarding the second point - i'm not sure. cc @ichub. my current feeling is that this is low enough priority for now to punt on.